shell/shared/admin.js

```diff
@@ -759,6 +856,49 @@ if (Meteor.isServer) {
       }
 
       return { sent: true };
+    },
+    offerIpNetwork: function (token) {
+      if (!isAdmin()) {
+        if (tokenIsValid(token)) {
+          throw new Meteor.Error(403, "Offering IpNetwork is only allowed for logged in users " +
+            "(a token is not sufficient). Please sign in with an admin account")
+        } else {
+          throw new Meteor.Error(403, "User must be admin.");
+        }
+      }
+
+      var requirements = [{
+        userIsAdmin: Meteor.userId()
+      }];
+      var sturdyRef = waitPromise(saveFrontendRef({ipNetwork: true}, {webkey: null},
+                                  requirements)).sturdyRef;
+      return ROOT_URL.protocol + "//" + makeWildcardHost("api") + "#" + sturdyRef;
+    },
+    offerIpInterface: function (token) {
+      if (!isAdmin()) {
+        if (tokenIsValid(token)) {
+          throw new Meteor.Error(403, "Offering IpInterface is only allowed for logged in users " +
+            "(a token is not sufficient). Please sign in with an admin account")
+        } else {
+          throw new Meteor.Error(403, "User must be admin.");
+        }
+      }
+
+      var requirements = [{
+        userIsAdmin: Meteor.userId()
+      }];
+      var sturdyRef = waitPromise(saveFrontendRef({ipInterface: true}, {webkey: null},
+                                  requirements)).sturdyRef;
+      return ROOT_URL.protocol + "//" + makeWildcardHost("api") + "#" + sturdyRef;
+    },
+    adminToggleDisableCap: function (token, capId, value) {
+      checkAuth(token);
+
+      if (value) {
+        ApiTokens.update({_id: capId}, {$set: {expires: new Date(0)}});
```

Comment:
Looks like your intent here is "soft" deletion. However, expired tokens get collected every hour:
https://github.com/sandstorm-io/sandstorm/blob/45c2ba13f56b6976ffb32a1a9eb38eab983a8516/shell/shared/grain.js#L130-L136